powersimdata/input/check.py

```diff
@@ -0,0 +1,195 @@
+import networkx as nx
+
+from powersimdata.input.grid import Grid
+
+
+def check_grid(grid):
+    """Check whether an object is an internally-consistent Grid object.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises TypeError: if ``grid`` is not a Grid object.
+    """
+    if not isinstance(grid, Grid):
+        raise TypeError("grid must be a Grid object")
+    _check_attributes(grid)
+    _check_for_islanded_buses(grid)
+    _check_for_undescribed_buses(grid)
+    _check_bus_against_bus2sub(grid)
+    _check_ac_interconnects(grid)
+    _check_transformer_substations(grid)
+    _check_line_voltages(grid)
+    _check_plant_against_gencost(grid)
+    _check_connected_components(grid)
+
+
+def _check_attributes(grid):
+    """Check whether a Grid object has the required attributes.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if ``grid`` is missing one or more required attributes.
+    """
+    required = {
+        "branch",
+        "bus",
+        "bus2sub",
+        "dcline",
+        "data_loc",
+        "gencost",
+        "grid_model",
+        "interconnect",
+        "model_immutables",
+        "plant",
+        "storage",
+        "sub",
+    }
+    for r in required:
+        if not hasattr(grid, r):
+            raise ValueError(f"grid object must have attribute {r}")
+
+
+def _check_for_islanded_buses(grid):
+    """Check whether a transmission network (AC & DC) does not connect to one or more
+    buses.
+
+    :param powersimdata.input.grid.Grid grid: grid or grid-like object to check.
+    :raises ValueError: if branches/DC lines exist, but one or more buses are islanded.
+    """
+    if len(grid.branch) + len(grid.dcline) > 0:
```

Comment:
Not sure whether we need this check or not.

Reply:
What if we have a single-bus Grid?

Reply:
Aha, then it is an island, isn't it? Yeah, you are right. We should keep this given we don't want this check to fail in that case.